src/ahocorasick.rs

```diff
@@ -523,6 +523,17 @@ impl<S: StateID> AhoCorasick<S> {
     ///     true
     /// });
     /// assert_eq!("0 the 2 to the 0age", result);
+    ///
+    /// let mut result = String::new();
```

Comment:
Could you add this as a separate example and add a small blurb about what it is demonstrating? (i.e., that returning `false` will short circuit the replacement.)